cpp/src/fil/internal.cuh

```diff
@@ -75,12 +75,13 @@ enum output_t {
 union val_t {
   /** threshold value for branch node or output value (e.g. class
       probability or regression summand) for leaf node */
-  float f;
-  /** class label */
+  float f = NAN;
+  /** class label, leaf vector index or categorical node set offset */
   int idx;
 };
 
 /** base_node contains common implementation details for dense and sparse nodes */
+template <bool can_be_categorical>
```

Comment:
Is there any real need to add this parameter to data structures as opposed to functions?

Reply:
as we discussed, since we're fine with losing 1 bit in sparse8 (and others for that matter), no real need.
just noting for posterity that we're aware of a workaround that restores the bit back.